pending-documentation.md

```diff
@@ -1321,6 +1414,43 @@ Note that in the case of Set and Map, one guarantee is that K1, K2, K3, and K4
 represent /distinct/ elements. Pattern matching fails if the correct number of
 distinct elements cannot be found.
 
+### Matching on cell fragments
+
+K allows matching fragments of the configuration and using them to construct
+terms and use as function parameters.
+
+```k
+configuration <t>
+                <k> #init ~> #collectOddFoos ~> $PGM </k>
+                <foos>
+                  <foo multiplicity="*" type="Set"> 1 </foo>
+                </foos>
+              </t>
+```
+
+The `#collectOddFoos` construct grabs the entire content of the `<foos>` cell.
+We may also match on only a portion of its content. Note that the fragment
+must be wrapped in a `<foo>` cell at the call site.
+
+```k
+syntax KItem ::= "#collectOddFoos"
+rule <k> #collectOddFoos => collectOddFoos(<foos> FOOS </foos>) ... </k>
+     <foos> FOOS </foos>
+```
+
+The `collectOddFoos` function collects the items it needs
+
+```k
+syntax Set ::= collectOddFoos(FoosCell) [function]
+rule collectOddFoos(<foos> <foo> I </foo> REST </foos>)
+  => SetItem(I) collectOddFoos(<foos> REST </foos>)
+  requires I %Int 2 ==Int 1
+rule collectOddFoos(<foos> <foo> I </foo> REST </foos>)
+  => collectOddFoos(<foos> REST </foos>)
+  requires I %Int 2 ==Int 0
+rule collectOddFoos(<foos> .Bag </foos>) => .Set
```

Comment:
```suggestion
    syntax Set ::= collectOddFoos(FoosCell) [function]
 // -----------------------------------------
    rule collectOddFoos(<foos> .Bag </foos>) => .Set
    rule collectOddFoos(<foos> <foo> I </foo> REST </foos>) => SetItem(I) collectOddFoos(<foos> REST </foos>) requires I %Int 2 ==Int 1
    rule collectOddFoos(<foos> <foo> I </foo> REST </foos>) => collectOddFoos(<foos> REST </foos>) requires I %Int 2 ==Int 0
```

This is just a suggestion, but try to format this one so it's more readable/scannable. I like to try and make it obvious what the differences between the multiple rules are.